src/NzbDrone.Core/Notifications/Trakt/TraktService.cs

```diff
@@ -0,0 +1,269 @@
+            };
+
+            var traktResolution = MapResolution(episodeFile.Quality.Quality.Resolution, episodeFile.MediaInfo?.ScanType);
+            var mediaType = MapMediaType(episodeFile.Quality.Quality.Source);
+            var audio = MapAudio(episodeFile);
+            var audioChannels = MapAudioChannels(episodeFile, audio);
+
+            var payloadEpisodes = new List<TraktEpisodeResource>();
+
+            foreach (var episode in episodeFile.Episodes.Value)
+            {
+                payloadEpisodes.Add(new TraktEpisodeResource
+                {
+                    Number = episode.EpisodeNumber,
+                    CollectedAt = DateTime.Now,
+                    Resolution = traktResolution,
+                    MediaType = mediaType,
+                    AudioChannels = audioChannels,
+                    Audio = audio,
+                });
+            }
+            
+            var payloadSeasons = new List<TraktSeasonResource>();
+            payloadSeasons.Add(new TraktSeasonResource
+            {
+                Number = episodeFile.SeasonNumber,
+                Episodes = payloadEpisodes
+            });
+
+            payload.Shows.Add(new TraktCollectShow
+            {
+                Title = series.Title,
+                Year = series.Year,
+                Ids = new TraktShowIdsResource
+                {
+                    Tvdb = series.TvdbId,
+                    Imdb = series.ImdbId ?? "",
+                },
+                Seasons = payloadSeasons,
+            }); ;
+
+            _proxy.AddToCollection(payload, settings.AccessToken);
+        }
+
+        private string MapMediaType(QualitySource source)
+        {
+            var traktSource = string.Empty;
+
+            switch (source)
+            {
+                case QualitySource.Web:
+                case QualitySource.WebRip:
+                    traktSource = "digital";
+                    break;
+                case QualitySource.BlurayRaw:
+                case QualitySource.Bluray:
+                    traktSource = "bluray";
+                    break;
+                case QualitySource.Television:
+                case QualitySource.TelevisionRaw:
+                case QualitySource.DVD:
+                    traktSource = "dvd";
+                    break;
+            }
+
+            return traktSource;
+        }
+
+        private string MapResolution(int resolution, string scanType)
+        {
+            var traktResolution = string.Empty;
+            var interlacedTypes = new string[] { "Interlaced", "MBAFF", "PAFF" };
+
+            var scanIdentifier = scanType.IsNotNullOrWhiteSpace() && interlacedTypes.Contains(scanType) ? "i" : "p";
+
+            switch (resolution)
+            {
+                case 2160:
+                    traktResolution = "uhd_4k";
+                    break;
+                case 1080:
+                    traktResolution = string.Format("hd_1080{0}", scanIdentifier);
```

Comment:
You can use string interpolation instead of `string.Format`:

`traktResolution = $"hd_1080{scanIdentifier}";`